### PR TITLE
rfr-16 Add Channel Items page. 

### PR DESCRIPTION
**Pull request by Andruhin**
Added all views and functionality for add/edit/remove/save/load channels. Calling channels list implemented on Settings Activity

 rfr-16 Add Channel Items page. Added calling channels list activity to Settings pref.

### CODE REVIEW

File: app/src/main/java/rss/feed/reader/managers/ChannelManager.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public static ChannelManager getInstance(){
+        return sInstance;
+    }
+
+    private String getJsonStringFromList(List<Channel> list){
+        return mGson.toJson(list);
+    }
+
+    private List<Channel> getChannelsListFromString(String jsonString){
+        List<Channel> list = new ArrayList<>(Arrays.asList(mGson.fromJson(jsonString, Channel[].class)));
```
Comment (fin1989) [inline app/src/main/java/rss/feed/reader/managers/ChannelManager.java]: return new ArrayList<>(Arrays.asList(mGson.fromJson(jsonString, Channel[].class)));

File: app/src/main/java/rss/feed/reader/ui/adapters/ChannelListAdapter.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.item_channel_list, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, final int position) {
```
Comment (fin1989) [inline app/src/main/java/rss/feed/reader/ui/adapters/ChannelListAdapter.java]: public void onBindViewHolder(ViewHolder holder, int position) {

File: app/src/main/java/rss/feed/reader/ui/adapters/ChannelListAdapter.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        final Channel channel = mList.get(position);
+        holder.title.setText(channel.getTitle());
+        holder.url.setText(channel.getUrl());
+        holder.position = position;
```
Comment (fin1989) [inline app/src/main/java/rss/feed/reader/ui/adapters/ChannelListAdapter.java]: holder.position = holder.getAdapterPosition();

File: app/src/main/java/rss/feed/reader/ui/activities/ChannelDetailsActivity.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import android.support.design.widget.CollapsingToolbarLayout;
+import android.support.v4.app.Fragment;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import rss.feed.reader.R;
+import rss.feed.reader.api.model.Channel;
+import rss.feed.reader.ui.fragments.channel_details.ChannelDetailsFragment;
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/ui/activities/ChannelDetailsActivity.java]: Please remove the underscore from the package name

File: app/src/main/java/rss/feed/reader/ui/adapters/ChannelListAdapter.java (hunk 3 above)
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/ui/adapters/ChannelListAdapter.java]: so no need to have position field ?

File: app/src/main/java/rss/feed/reader/ui/fragments/channel_list/ChannelListContract.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.util.List;
+
+import rss.feed.reader.api.model.Channel;
+
+/**
+ * Created by Andriy Ksenych on 03.10.2016.
+ */
+
+public interface ChannelListContract {
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/ui/fragments/channel_list/ChannelListContract.java]: why do you need it?

File: app/src/main/res/layout/activity_channel_list.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/activity_channel_list"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="rss.feed.reader.ui.activities.ChannelListActivity">
+
+<FrameLayout
```
Comment (Vovchyk) [inline app/src/main/res/layout/activity_channel_list.xml]: padding

File: app/src/main/java/rss/feed/reader/Navigation.java [hunk 7]
```diff
@@ -25,6 +30,26 @@ public static void toWebBrowser(@NonNull ActivityStarter starter, @NonNull Strin
...
+    }
+
+    /**
+     * Navigates to channel detail activity
+     */
+    public static void toChannelDetails(@NonNull ActivityStarter starter, int position, @Nullable Channel channel) {
+        Intent intent = starter.createIntent(ChannelDetailsActivity.class);
+        intent.putExtra(ChannelDetailsActivity.EXTRAS_CHANNEL, channel);
+        intent.putExtra(ChannelDetailsActivity.EXTRAS_POSITION, position);
+        starter.startActivityForResult(intent, ChannelDetailsActivity.ACTIVITY_CHANNEL_DETAILS_RESULT);
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/Navigation.java]: Strange name. I suppose ACTIVITY_CHANNEL_DETAILS_RESULT is a request code?.. Also i would keep it as part of this class

File: app/build.gradle [hunk 8]
```diff
@@ -22,12 +22,14 @@ android {
 
 ext {
     supportLibraryVersion = '24.2.0'
+    designLibraryVersion = '24.2.0'
```
Comment (qUser13) [inline app/build.gradle]: Can we merge design and supportLibrary versions?

File: app/src/main/java/rss/feed/reader/ui/activities/ChannelListActivity.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+import rss.feed.reader.R;
+import rss.feed.reader.ui.fragments.channel_list.ChannelListFragment;
+
+/**
+ * Created by Andriy Ksenych on 17.10.2016.
+ * Activity for showing user's channels list.
+ */
+
+public class ChannelListActivity extends AppCompatActivity {
```
Comment (qUser13) [inline app/src/main/java/rss/feed/reader/ui/activities/ChannelListActivity.java]: Please, use BaseActivity as parent

File: app/build.gradle (hunk 8 above)
Comment (Andruhin) [inline app/build.gradle]: We can, but sometimes one of them can be updated sooner that another one.

File: app/src/main/java/rss/feed/reader/ui/adapters/ChannelListAdapter.java (hunk 3 above)
Comment (Andruhin) [inline app/src/main/java/rss/feed/reader/ui/adapters/ChannelListAdapter.java]: yes. I'll remove it from onBindViewHolder method

File: app/src/main/java/rss/feed/reader/managers/ChannelManager.java (hunk 1 above)
Comment (qUser13) [inline app/src/main/java/rss/feed/reader/managers/ChannelManager.java]: You will make an array, then make a list and then make a copy of the previous list. Better check https://sites.google.com/site/gson/gson-user-guide#TOC-Collections-Examples

Comment (Andruhin) [inline app/src/main/java/rss/feed/reader/managers/ChannelManager.java]: thanks! It works pretty well!

File: app/src/main/res/layout/activity_channel_details.xml [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        <android.support.design.widget.CollapsingToolbarLayout
+            android:id="@+id/activity_channel_details_collapsing_toolbar_layout"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:fitsSystemWindows="true"
+            app:contentScrim="?attr/colorPrimary"
+            app:layout_scrollFlags="scroll|exitUntilCollapsed"
+            android:background="@color/colorPrimaryDark">
+
+            <de.hdodenhof.circleimageview.CircleImageView
```
Comment (qUser13) [inline app/src/main/res/layout/activity_channel_details.xml]: Avoid hard-coded dimensions

File: app/src/main/res/layout/activity_channel_details.xml [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+                android:layout_height="?attr/actionBarSize"
+                app:layout_collapseMode="pin"
+                app:popupTheme="@style/AppTheme.PopupOverlay">
+
+            </android.support.v7.widget.Toolbar>
+
+        </android.support.design.widget.CollapsingToolbarLayout>
+    </android.support.design.widget.AppBarLayout>
+
+    <android.support.v4.widget.NestedScrollView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (qUser13) [inline app/src/main/res/layout/activity_channel_details.xml]: Cleanup duplicated annotations

**Pull request closed without merging**